src/java/org/apache/cassandra/config/Converters.java

```diff
@@ -36,26 +36,29 @@
      * This converter is used when we change the name of a cassandra.yaml configuration parameter but we want to be
      * able to still use the old name too. No units involved.
      */
-    IDENTITY(null, null, o -> o, o-> o),
-    MILLIS_DURATION(Long.class, SmallestDurationMilliseconds.class,
-                    SmallestDurationMilliseconds::inMilliseconds,
-                    o -> o.toMilliseconds()),
+    IDENTITY(null, null, o -> o, o -> o),
+    MILLIS_DURATION_LONG(Long.class, SmallestDurationMilliseconds.class,
+                         SmallestDurationMilliseconds::inMilliseconds,
+                         o -> o.toMilliseconds()),
+    MILLIS_DURATION_INT(Integer.class, SmallestDurationMilliseconds.class,
+                        i -> SmallestDurationMilliseconds.inMilliseconds(i),
+                        DurationSpec::toMillisecondsAsInt),
     MILLIS_DOUBLE_DURATION(Double.class, SmallestDurationMilliseconds.class,
```

Comment:
nit: I guess we should make this `MILLIS_DURATION_DOUBLE` to be consistent w/ the other two?